x/provider/types/msgs_test.go

```diff
@@ -24,17 +24,17 @@ func TestConfigPath(t *testing.T) {
 			path:   "foo.yml",
 			expErr: ErrNotAbsProviderURI,
 		},
-		/*{
-			path:   "localhost:80/foo", // would expect this to cause error, but it does not.
-			expErr: ErrNotAbsProviderURI,
-		},*/
+		{
+			path:   "localhost:80/foo",
+			expErr: ErrInvalidProviderURI,
+		},
 		{
 			path:   "file:///foo.yml",
-			expErr: nil,
+			expErr: ErrInvalidProviderURI,
 		},
 		{
 			path:   "http://localhost:80/foo",
-			expErr: nil,
+			expErr: ErrInvalidProviderURI,
```

Comment:
I think we need to add some more tests here.  Can you make sure the relevant entries from the table in [this comment](https://github.com/ovrclk/akash/pull/612#discussion_r428983631) are tested?

Reply:
Wouldn't regex be easier? https://regex101.com/r/Xc2w0v/2/